src/plugins/credential_management/server/saved_objects/credential_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  HttpServiceStart,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from 'opensearch-dashboards/server';
+//   import { SavedObjectsErrorHelpers } from '../../../../src/core/server';
+import _ from 'lodash';
+import { handleEncryption } from '../credential_manager';
+
+export class CredentialSavedObjectsClientWrapper {
+  public httpStart?: HttpServiceStart;
+
+  constructor() {}
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const createWithCredentialMaterialsContentEncryption = async <T = unknown>(
+      type: string,
+      attributes: T,
+      options?: SavedObjectsCreateOptions
+    ) => {
+      // TODO: Add validation
+      const encryptedAttributes = {
+        title: attributes.title,
+        description: attributes.description,
+        credentialType: attributes.credentialType,
+        credentialMaterials: await handleEncryption(attributes.credentialMaterials),
+      };
+      return await wrapperOptions.client.create(type, encryptedAttributes, options);
+    };
+
```

Comment:
Curious to know if we going to provide a getAndDecrypt API to explicitly Decrypt the password?

Reply:
Discussed here https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1926